Source/WebCore/platform/graphics/SourceBufferPrivate.cpp

```diff
@@ -100,10 +94,13 @@ void SourceBufferPrivate::updateHighestPresentationTimestamp()
         m_client->sourceBufferPrivateHighestPresentationTimestampChanged(m_highestPresentationTimestamp);
 }
 
-void SourceBufferPrivate::setBufferedRanges(const PlatformTimeRanges& timeRanges)
+void SourceBufferPrivate::setBufferedRanges(PlatformTimeRanges&& timeRanges)
 {
-    m_buffered->ranges() = timeRanges;
-    setBufferedDirty(true);
+    if (m_buffered->ranges() == timeRanges)
```

Comment:
This is a pre-existing issue, but we should change `RefPtr<PlatformTimeRanges> m_buffered` -> `Ref<PlatformTimeRanges>`.

Reply:
done

Reply:
actually, I'll make the buffered() method return a reference instead and have the member be a Ref<T>. but it's a much bigger change. Will do so in a follow up change